src/components/views/rooms/LinkPreviewWidget.tsx

```diff
@@ -125,7 +125,8 @@ export default class LinkPreviewWidget extends React.Component<IProps> {
 
         // Youtube video player embed
         const youtubeRegex = /^https?:\/\/(m[.]|www[.])?(youtube[.]com\/watch[?]v=|youtu[.]be\/)([\w-]+)(\S+)?$/;
-        if (this.props.youtubeEmbedPlayerEnabled && this.props.link.match(youtubeRegex)) {
+
+        if (this.props.youtubeEmbedPlayerEnabled || SettingsStore.getValue("youtubeEmbedPlayer") && this.props.link.match(youtubeRegex)) {
```

Comment:
This change looks very weird. What's the reason for it failing. Something tells me this is not the right way to solve it.

Reply:
this.props.youtubeEmbedPlayerEnabled gets set to setting value only when its updated, it is undefined by default.